examples/calling-apis/spa-with-backend-api/react-hono-ai-sdk/server/src/index.ts

```diff
@@ -0,0 +1,168 @@
+import "dotenv/config";
+
+import {
+  createDataStreamResponse,
+  generateId,
+  streamText,
+  ToolExecutionError,
+} from "ai";
+import { Hono } from "hono";
+import { cors } from "hono/cors";
+import { decodeJwt } from "jose";
+
+import { openai } from "@ai-sdk/openai";
+import { setAIContext } from "@auth0/ai-vercel";
+import {
+  InterruptionPrefix,
+  withInterruptions,
+} from "@auth0/ai-vercel/interrupts";
+import { Auth0Interrupt } from "@auth0/ai/interrupts";
+import { serve } from "@hono/node-server";
+
+import { listNearbyEvents } from "./lib/tools/listNearbyEvents";
+import { listUserCalendars } from "./lib/tools/listUserCalendars";
+import { jwtAuthMiddleware } from "./middleware/auth";
+
+import type { ApiResponse } from "shared/dist";
+
+const getAllowedOrigins = (): string[] => {
+  const allowedOrigins = process.env.ALLOWED_ORIGINS;
+  if (!allowedOrigins) {
+    // Fallback to default origins if not set
+    return ["http://localhost:5173", "http://localhost:3000"];
+  }
+  return allowedOrigins.split(",").map((origin) => origin.trim());
+};
+
+export const app = new Hono()
+
+  .use(
+    cors({
+      origin: getAllowedOrigins(),
+      allowHeaders: ["Content-Type", "Authorization"],
+      allowMethods: ["GET", "POST", "PUT", "DELETE", "OPTIONS"],
+    })
+  )
+
+  .get("/", (c) => {
+    return c.text("Hello Hono!");
+  })
+
+  .get("/hello", async (c) => {
+    const data: ApiResponse = {
+      message: "Hello BHVR!",
+      success: true,
+    };
+    console.log("✅ Success! Public /hello route called!");
+    return c.json(data, { status: 200 });
+  })
+
+  // Protected API route
+  .get("/api/external", jwtAuthMiddleware(), async (c) => {
+    const user = c.get("user");
+
+    // Extract and log the access token
+    const authHeader = c.req.header("authorization");
+    const accessToken = authHeader?.replace("Bearer ", "");
+
+    // Decode and log the JWT payload
+    if (accessToken) {
+      try {
+        const decodedJwt = decodeJwt(accessToken);
+        console.log("🔓 Decoded JWT:", JSON.stringify(decodedJwt, null, 2));
+      } catch (error) {
+        console.error("❌ Error decoding JWT:", error);
+      }
+    }
+
+    const data: ApiResponse = {
+      message: `Your access token was successfully validated! Welcome ${user.sub}`,
+      success: true,
+    };
+
+    return c.json(data, { status: 200 });
+  })
+
+  .post("/chat", jwtAuthMiddleware(), async (c) => {
+    const user = c.get("user");
```

Comment:
planning to move this endpoint implementation to it's own file. Just seeing it will make the docs-v2 reference a lot cleaner (smaller and more targeted).

Reply:
Actually, seems ok to me... have just abbreviated it some for now in the documentation snippet for now.